### DISCUSSION TITLE
A File Structure As Template

### DISCUSSION THREAD

**Discussion by TacticalGator** [SilentVoid13/Templater#987] [Help]
Hey all

I was hoping if any of you could help me.

The issue here is that I need to use a whole file structure as a template.

![image](https://user-images.githubusercontent.com/37283676/211209269-95755915-912c-4a41-ad22-259dc1e81816.png)
This is just an example above. 
I want to use this structure as a template

![image](https://user-images.githubusercontent.com/37283676/211210046-72e300d9-deff-47d0-b65b-d5d48db53353.png)
I was thinking there should be..
A MD template with the commands that recursively duplicates the structure above and pastes on to the current working directory. either that or to the vault root.

![image](https://user-images.githubusercontent.com/37283676/211209506-d1b3ea21-04f9-40b3-8253-c2fa2d14fb1d.png)
So it can be called from Ctrl + N

I've been using a PowerShell script to get the job done.
I just think it would be better if I could get it done within the app itself with this plugin. ( I know it's totally possible)

I looked up pretty much everywhere. Here, obsidian forum, reddit, etc.

Some of the posts that I saw are rather relevant:
https://forum.obsidian.md/t/duplicate-folder-from-inside-obsidian/29370
https://github.com/SilentVoid13/Templater/discussions/569
https://forum.obsidian.md/t/copy-folders-from-one-location-to-other/46615

Some of them cover the exact same issue that I am having.

I tried doing it myself for hours without any luck as I am not familiar with JS and I really don't want to get to the system commands

Please help me out

**Comment by welpdx** [comment]
Can you help me understand exactly what you want?
Is the following example close to getting what you want?

Say you are at 

myVault/
├─ folder1/
│  ├─ folder2/

at folder2 you run the script and it creates an arrangement of folders like so:

myVault/
├─ folder1/
│  ├─ folder2/
│  │  ├─ structure/
│  │  │  ├─ generatedFolder1/
│  │  │  ├─ generatedFolder2/
│  │  │  ├─ generatedFolder3/


Is this right?

Do you also want the script to generate files in the generatedFolders? 
Do you also want a template to generate content in those files? 
Is files1 files2 and files3 the same in each generatedFolder?


myVault/
├─ folder1/
│  ├─ folder2/
│  │  ├─ structure/
│  │  │  ├─ generatedFolder1/
│  │  │  │  ├─ file1
│  │  │  │  ├─ file2
│  │  │  │  ├─ file3
│  │  │  ├─ generatedFolder2/
│  │  │  │  ├─ file1
│  │  │  │  ├─ file2
│  │  │  │  ├─ file3
│  │  │  ├─ generatedFolder3/
│  │  │  │  ├─ file1
│  │  │  │  ├─ file2
│  │  │  │  ├─ file3
